sdk/search/Azure.Search.Documents/src/SearchServiceClient.cs

```diff
@@ -1088,24 +1276,36 @@ public virtual SearchClient GetSearchClient(string indexName)
         /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
         /// <returns>The <see cref="Pageable{T}"/> from the server containing a list of <see cref="SearchIndex"/> names.</returns>
         /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
-        [ForwardsClientCalls]
         public virtual Pageable<string> GetIndexNames(
             SearchRequestOptions options = null,
-            CancellationToken cancellationToken = default) => PageResponseEnumerator.CreateEnumerable((continuationToken) =>
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = ClientDiagnostics.CreateScope($"{nameof(SearchServiceClient)}.{nameof(GetIndexNames)}");
+            scope.Start();
+            try
             {
-                if (continuationToken != null)
+                return PageResponseEnumerator.CreateEnumerable((continuationToken) =>
                 {
-                    throw new NotSupportedException("A continuation token is unexpected and unsupported at this time.");
-                }
+                    if (continuationToken != null)
+                    {
+                        throw new NotSupportedException("A continuation token is unexpected and unsupported at this time.");
```

Comment:
This is kind of a confusing error message (because what time should a user be passing the continuation token then?).  Probably less confusing to just say it's not supported.